mmv1/third_party/terraform/tests/resource_network_services_gateway_test.go.erb

```diff
@@ -67,4 +67,670 @@ resource "google_network_services_gateway" "foobar" {
+  ip_cidr_range = "192.168.0.0/23"
+  region        = "us-east1"
+  network       = google_compute_network.default.id
+  role          = "ACTIVE"
+}
+
+resource "google_compute_subnetwork" "default" {
+  name          = "%s"
+  purpose       = "PRIVATE"
+  ip_cidr_range = "10.128.0.0/20"
+  region        = "us-east1"
+  network       = google_compute_network.default.id
+  role          = "ACTIVE"
+}
+
+resource "google_network_security_gateway_security_policy" "default" {
+  name        = "%s"
+  location    = "us-east1"
+}
+
+resource "google_network_security_gateway_security_policy_rule" "default" {
+  name                    = "%s"
+  location                = "us-east1"
+  gateway_security_policy = google_network_security_gateway_security_policy.default.name
+  enabled                 = true  
+  priority                = 1
+  session_matcher         = "host() == 'example.com'"
+  basic_profile           = "ALLOW"
+}
+
+# TODO(#14600): this field will be updatable soon so this test should also cover it.
+# resource "google_network_security_gateway_security_policy" "newpolicy" {
+#   name        = "%s"
+#   location    = "us-east1"
+# }
+
+# resource "google_network_security_gateway_security_policy_rule" "newrule" {
+#   name                    = "%s"
+#   location                = "us-east1"
+#   gateway_security_policy = google_network_security_gateway_security_policy.newpolicy.name
+#   enabled                 = true  
+#   priority                = 1
+#   session_matcher         = "host() == 'example.com'"
+#   basic_profile           = "ALLOW"
+# }
+	  
+resource "google_network_services_gateway" "foobar" {
+  name                                 = "%s"
+  location                             = "us-east1"
+  addresses                            = ["10.128.0.99"]
+  type                                 = "SECURE_WEB_GATEWAY"
+  ports                                = [443]
+  description                          = "updated description"
+  scope                                = "%s"
+  certificate_urls                     = [google_certificate_manager_certificate.default.id, google_certificate_manager_certificate.newcm.id]
+  gateway_security_policy              = google_network_security_gateway_security_policy.default.id
+  # TODO(#14600): this field will be updatable soon so this test should also cover it.
+  # gateway_security_policy              = google_network_security_gateway_security_policy.newpolicy.id
+  network                              = google_compute_network.default.id
+  subnetwork                           = google_compute_subnetwork.default.id
+  delete_swg_autogen_router_on_destroy = true
+  depends_on                           = [google_compute_subnetwork.proxyonlysubnet]
+
+}
+`, cmName, newCmName, netName, subnetName, pSubnetName, policyName, newPolicyName, ruleName, newRuleName, gatewayName, gatewayScope)
+}
+
+func TestAccNetworkServicesGateway_multipleSwpGatewaysDifferentSubnetwork(t *testing.T) {
```

Comment:
checked the debug logs for `TestAccNetworkServicesGateway_multipleSwpGatewaysDifferentSubnetwork` and `TestAccNetworkServicesGateway_multipleSwpGatewaysDifferentNetwork`. It seems that deletion works as intended.

Reply:
Great. Thanks.